pkg/minikube/bootstrapper/bsutil/kubeadm.go

```diff
@@ -88,6 +85,15 @@ func GenerateKubeadmYAML(cc config.ClusterConfig, n config.Node, r cruntime.Mana
 
 	// ref: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration
 	kubeletConfigOpts := kubeletConfigOpts(k8s.ExtraOptions)
+	// container-runtime-endpoint kubelet flag was deprecated but corresponding containerRuntimeEndpoint kubelet config field is "required" but supported only from k8s v1.27
+	// ref: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/#options
+	// ref: https://github.com/kubernetes/kubernetes/issues/118787
+	if version.GTE(semver.MustParse("1.27.0")) {
+		kubeletConfigOpts["containerRuntimeEndpoint"] = k8s.ExtraOptions.Get("container-runtime-endpoint", Kubelet)
+		if kubeletConfigOpts["containerRuntimeEndpoint"] == "" {
+			kubeletConfigOpts["containerRuntimeEndpoint"] = r.KubeletOptions()["container-runtime-endpoint"]
+		}
```

Comment:
```suggestion
		runtimeEndpoint := k8s.ExtraOptions.Get("container-runtime-endpoint", Kubelet)
		if runtimeEndpoint == "" {
			runtimeEndpoint = r.KubeletOptions()["container-runtime-endpoint"]
		}
		kubeletConfigOpts["containerRuntimeEndpoint"] = runtimeEndpoint
```
Use variable to shorten long lines